File: Python/8py 1/ball.py
"""
|--------------------------------------------------------------------------
| Regular Ball Super Ball
|--------------------------------------------------------------------------
|
"""

class Ball:
   def __init__(self, ball_type = "regular"):
      self.ball_type = ball_type


class Ball:
   __init__ = lambda _,a=0: setattr(Ball,'ball_type',('regular','super')[bool(a)])


class Ball(object):
   def __init__(self, type_ = "regular"):
      self.ball_type = type_


print(Ball().ball_type, "regular")
print(Ball('super').ball_type, "super")

File: JavaScript/8js 1/ball.js
/*
|--------------------------------------------------------------------------
| Regular Ball Super Ball
|--------------------------------------------------------------------------
|
*/

class Ball {
   constructor(ballType = "regular") {
      this.ballType = ballType;
   }
}


const Ball = function (ballType = "regular") {
   this.ballType = ballType;
};


const Ball = function (ballType) {
   this.ballType = ballType || "regular";
};


console.log(new Ball().ballType, "regular");
console.log(new Ball("super").ballType, "super");